webpify.js

```diff
@@ -0,0 +1,184 @@
+const sharp = require('sharp');
+const glob = require('glob');
+const path = require('path');
+const fs = require('fs').promises;
+
+// Array to store original file paths
+const originalFiles = [];
+const convertedFiles = [];
+
+async function deleteOriginalFiles() {
+    console.log('\nDeleting original files...');
+    for (const conversion of convertedFiles) {
+        if (conversion.status === 'converted' || conversion.status === 'skipped') {
+            try {
+                await fs.unlink(conversion.original);
+                console.log(`Deleted: ${conversion.original}`);
+            } catch (error) {
+                console.error(`Error deleting ${conversion.original}:`, error);
+            }
+        }
+    }
+    console.log('Original file deletion complete!');
+}
+
+async function replaceInFiles() {
+    // Find all JS, JSX, TS, TSX, CSS, SCSS, HTML files, excluding webpify.js and pre-render.js
+    const codeFiles = glob.sync('**/*.{js,jsx,ts,tsx,css,scss,html}', {
+        ignore: [
+            'node_modules/**',
+            'build/**',
+            'dist/**',
+            'webpify.js',
+            'pre-render.js'
+        ]
+    });
+
+    for (const filePath of codeFiles) {
+        try {
+            let content = await fs.readFile(filePath, 'utf8');
+            let hasChanges = false;
+
+            // Replace each image reference
+            for (const conversion of convertedFiles) {
+                const { original, webp } = conversion;
+                const fileName = path.basename(original);
+                const webpFileName = path.basename(webp);
+
+                if (original.includes('public/')) {
+                    // Create the absolute path version (starting with /)
+                    const absolutePath = `/${original.split('public/')[1]}`;
+                    const absoluteWebPPath = `/${webp.split('public/')[1]}`;
```

Comment:
If the path does not include "public/", then we know not to check if references for it include %PUBLIC_URL%, or start with just /, because... They're not in the publix folder. Anything that fails this check, are images in src/, which follow the rest of the code.